components/ntp_sponsored_images/browser/ntp_sponsored_images_service.cc

```diff
@@ -0,0 +1,197 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_images_service.h"
+
+#include <algorithm>
+
+#include "base/bind.h"
+#include "base/files/file_util.h"
+#include "base/json/json_reader.h"
+#include "base/task/post_task.h"
+#include "base/values.h"
+#include "brave/components/brave_ads/browser/locale_helper.h"
+#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_images_component_installer.h"
+#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_images_data.h"
+#include "brave/components/ntp_sponsored_images/browser/ntp_sponsored_images_internal_data.h"
+#include "brave/components/ntp_sponsored_images/browser/regional_component_data.h"
+#include "brave/components/ntp_sponsored_images/browser/switches.h"
+#include "brave/vendor/bat-native-ads/src/bat/ads/internal/locale_helper.h"
```

Comment:
this header is internal and no header should be accessed using `#include "brave/vendor...`